packages/chain-adapters/src/utxo/UTXOBaseAdapter.ts

```diff
@@ -0,0 +1,159 @@
+import { CAIP2, caip2, caip19 } from '@shapeshiftoss/caip'
+import { bip32ToAddressNList, HDWallet, PublicKey } from '@shapeshiftoss/hdwallet-core'
+import { BIP44Params, chainAdapters, ChainTypes, UtxoAccountType } from '@shapeshiftoss/types'
+import { bitcoin } from '@shapeshiftoss/unchained-client'
+import WAValidator from 'multicoin-address-validator'
+
+import { ChainAdapter as IChainAdapter } from '../api'
+import { ErrorHandler } from '../error/ErrorHandler'
+import {
+  accountTypeToScriptType,
+  bnOrZero,
+  convertXpubVersion,
+  toRootDerivationPath
+} from '../utils'
+
+export type UTXOChainTypes = ChainTypes.Bitcoin // to be extended in the future to include other UTXOs
+
+export interface ChainAdapterArgs {
+  providers: {
+    http: bitcoin.api.V1Api //?
+    ws: bitcoin.ws.Client
+  }
+  coinName: string
+  chainId?: CAIP2
+}
+
+/**
+ * Base chain adapter for all UTXO chains. When extending please add your ChainType to the
+ * UTXOChainTypes. For example:
+ *
+ * `export type UTXOChainTypes = ChainTypes.Bitcoin | ChainTypes.Litecoin`
+ */
+export abstract class UTXOBaseAdapter<T extends UTXOChainTypes> implements IChainAdapter<T> {
+  protected chainId: CAIP2
+  protected coinName: string
+  protected readonly providers: {
+    http: bitcoin.api.V1Api
+    ws: bitcoin.ws.Client
+  }
+
+  public static readonly defaultBIP44Params: BIP44Params = {
+    purpose: 84, // segwit native
+    coinType: 0,
+    accountNumber: 0
+  }
+
+  protected constructor(args: ChainAdapterArgs) {
+    this.providers = args.providers
+  }
+
+  /* Abstract Methods */
+
+  abstract subscribeTxs(
+    input: chainAdapters.SubscribeTxsInput,
+    onMessage: (msg: chainAdapters.SubscribeTxsMessage<T>) => void,
+    onError?: (err: chainAdapters.SubscribeError) => void
+  ): Promise<void>
+  abstract unsubscribeTxs(input?: chainAdapters.SubscribeTxsInput): void
+  abstract closeTxs(): void
+  abstract getType(): T
+
+  abstract getTxHistory(
+    input: chainAdapters.TxHistoryInput
+  ): Promise<chainAdapters.TxHistoryResponse<T>>
+
+  abstract buildSendTransaction(
+    tx: chainAdapters.BuildSendTxInput<T>
+  ): Promise<{ txToSign: chainAdapters.ChainTxType<T> }>
+
+  abstract getAddress(input: chainAdapters.GetAddressInput): Promise<string>
+
+  abstract getFeeData(
+    input: Partial<chainAdapters.GetFeeDataInput<T>>
+  ): Promise<chainAdapters.FeeDataEstimate<T>>
+
+  abstract signTransaction(
+    signTxInput: chainAdapters.SignTxInput<chainAdapters.ChainTxType<T>>
+  ): Promise<string>
+
+  /* public methods */
+
+  getCaip2(): CAIP2 {
+    return this.chainId
+  }
+
+  getChainId(): CAIP2 {
+    return this.chainId
+  }
+
+  async getAccount(pubkey: string): Promise<chainAdapters.Account<T>> {
```

Comment:
this may also need to move into the implementing class. `this.providers.http.getAccount` for BCH for example will end up calling a different coinstack at like `api.doge.shapeshift.com` - @kaladinlight can confirm

Reply:
Yeah it seems like we need to think through this abstraction (both cosmos and utxo) a bit more to ensure we are setting this up correctly.

Reply:
if we had a generic provider interface, this would not matter which coinstack it ended up calling, right?

Reply:
@kaladinlight - can you confirm that if we had a proper abstraction on the provider this would be fine to leave in the base class?

Reply:
Each http provider is generated directly from the swagger.json being served by the specific coinstack. There is BaseAPI interface that each coinstack api must implement, but the base return payloads are very minimal and intended to be extended by each coinstack as they see fit. We also do not expose this interface anywhere at this time.

We could try and create this interface as a part of the client package to use within chain adapters. I'm also wondering if we could utilize a type union and that would just result in the shared result set. We do need to think about what it would look like to extend the base logic with any possible unique data for a specific child chain adapter.

Reply:
Okay, for this PR is your recommendation then to move this to the implementing class until we have a more generic solution?

Reply:
I think you can just leave it for now and make a comment that when additional utxo chains are added, we need to ensure the generic base implementation works against all imported api types. We only support BTC anyways right now and this is the same setup that cosmos base adapter currently has.

Reply:
awesome, thanks will do and move this to a "non-draft" pr so we can get it merged.